### DISCUSSION TITLE
How about use `ast-grep` instead of `regex`  to transform attributes in jsx

### DISCUSSION THREAD

**Discussion by zhiyuanzmj** [unocss/unocss#3346] [Ideas]
Now `transformerAttributifyJsx` using a complex regex to match attributes. It have many bug that I tried create a lot of PR to resolve, But new bugs are always found, such as: 

``` tsx
<div flex>12</div>
```
Above of the line is not work  because of the regex not matched, and I don't know how to resolve.
https://github.com/unocss/unocss/blob/efdc358897a308323e1d943dd0f7c13e344e1495/packages/transformer-attributify-jsx/src/index.ts#L40-L42

### Use ast-grep
So, I created the repo [transformer-attributify-jsx-sg](https://github.com/zhiyuanzmj/transformer-attributify-jsx-sg).\
It's using `ast-grep` to match attributes, so have much more readable, simple and also fast.
```ts
const nodes = root.findAll({
  rule: {
    kind: 'jsx_attribute',
    not: {
      has: {
        any: [{ kind: 'jsx_expression' }, { kind: 'string' }],
      },
    },
  },
})
```

You can use `pnpm run bench` to benchmark test:
```
Running "unocss concurrent transform jsx" suite...
benchRegexTransform:
  16 ops/s, ±9.95%   | fastest

benchAstGrepTransform:
  13 ops/s, ±0.34%   | 18.75% slower

benchSwcTransform:
  6 ops/s, ±0.84%    | 62.5% slower

benchBabelTransform:
  2 ops/s, ±1.44%    | slowest, 87.5% slower
```
If you agree I will create PR.

**Comment by HerringtonDarkholme** [comment]
Hey! ast-grep author here.
I'm flattered to see that unocss is interested in using ast-grep! 
Let me know if you have any questions and I will also try improving the runtime performance of napi.

That said, I respect unocss' decision and understand your constraints on choosing library. 

Best!
Herrington
